presto-raptor/src/main/java/com/facebook/presto/raptor/storage/ShardEjector.java

```diff
@@ -242,8 +253,8 @@ void process()
             shardManager.replaceShardAssignment(shard.getTableId(), shardUuid, target, false);
 
             // delete local file
-            File file = storageService.getStorageFile(shardUuid);
-            if (file.exists() && !file.delete()) {
+            Path file = storageService.getStorageFile(shardUuid);
```

Comment:
Why don't we just convert `Path` to `File` and use OS file system call like what we did in `BackupManager`?

Reply:
I think in more generic classes like `OrcPageRewriter` and `OrcStorageManager` we can stick to using `Path`. But for local flash specific components like `BackupManager`, `ShardEjector`, `ShardRecoveryManager` and `ShardCleaner` maybe we can always convert `Path` to `File` when invoking file system related call.

Reply:
Two reasons:
- To work on `File`, we need to call `RawLocalFileSystem::pathToFile` to convert `Path` to `File`. That requires an explicit file system anyway.
- Also, it provides better abstraction.

Reply:
I am not very strong on this. Here is my thoughts:

For classes like `OrcPageRewriter` and `OrcStorageManager`, they will be used by both local and disagg storage. So they have to operate on hdfs `Path`, not `File`. But for local storage specific classes (`BackupManager`, `ShardEjector`, `ShardRecoveryManager` and `ShardCleaner`), they anyway need a local file system to operate on. Using `File` and calling OS file system call (i.e. `file.delete()`, `file.exists()`) makes sure their behavior is exactly the same as previous and introduces minimum changes.